pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -465,7 +443,7 @@ protected void handleGetLastMessageIdSuccess(CommandGetLastMessageIdResponse suc
             log.debug("{} Received success GetLastMessageId response from server: {}", ctx.channel(), success.getRequestId());
         }
         long requestId = success.getRequestId();
-        CompletableFuture<MessageIdData> requestFuture = pendingGetLastMessageIdRequests.remove(requestId);
+        CompletableFuture<MessageIdData> requestFuture = (CompletableFuture<MessageIdData>) pendingRequests.remove(requestId);
```

Comment:
if there is a cast exception, shouldn't we handle it?  If the client somehow is returned a wrong request id and causing cast exception here, I think the requestFuture will never get completed and may cause the client api call to block forever.

Reply:
The connection will get automatically closed and therefore the future will be completed there.

Reply:
i see